keras_cv/callbacks/pycoco_callback.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import six
+import tensorflow as tf
+from keras.callbacks import Callback
+
+from keras_cv import bounding_box
+from keras_cv.metrics.coco import compute_pycoco_metrics
+
+
+class PyCOCOCallback(Callback):
+    def __init__(self, bounding_box_format):
+        self.model = None
+        self.bounding_box_format = bounding_box_format
+        super().__init__()
+
+    def set_model(self, model):
+        self.model = model
+
+    def on_epoch_end(self, epoch, logs=None):
+        logs = logs or {}
+
+        dataset = self.model._eval_data_handler._dataset
+        gt = {}
+        preds = {}
+
+        for i, batch in enumerate(dataset):
+            gt_i, preds_i = self._eval_batch(batch, i)
+
+            for k, v in six.iteritems(preds_i):
+                if k not in preds:
+                    preds[k] = [v]
+                else:
+                    preds[k].append(v)
+
+            for k, v in six.iteritems(gt_i):
+                if k not in gt:
+                    gt[k] = [v]
+                else:
+                    gt[k].append(v)
+
+        metrics = compute_pycoco_metrics(gt, preds)
+        # Mark these as validation metrics by prepending a val_ prefix
+        metrics = {"val_" + name: val for name, val in metrics.items()}
+
+        logs.update(metrics)
+
+    def _eval_batch(self, batch, index):
+        images, gt_boxes = batch
```

Comment:
I think we should only make sure this works with retinanet.fit and fasterrcnn.fit at this moment, so accepting boxes and classes separately is fine.

Reply:
retinanet.fit assumes that boxes and classes are together in one tensor. This currently works with retinanet.fit but not fasterrcnn.fit. If I changed it, that would flip flop. More discussion in the thread above.

Reply:
Then it seems we should make sure this works with fasterrcnn.fit not retinanet.fit -- given we have discussed this several times and it seems that we have a consensus that we should separate them out, I'd prefer not introduce more tech debt.

Reply:
Fine by me -- I will update this to work with fasterrcnn, not retinanet.

Once that's done, and I've added test, I'll re-request review.

Reply:
I believe we settled on a dictionary format, maybe it’s time migrate all components to the dictionary format for bounding boxes.